### DISCUSSION TITLE
GitLab Merge Request Integration on HTTP GitLab Instances

### DISCUSSION THREAD

**Discussion by pschneider** [WeblateOrg/weblate#9161] [Q&A] [answered]
Hey,

we're looking into using the `GitLab Merge Request` integration as version control setting on our self-managed GitLab instance. This GitLab instance is only available internally and due to some historical setup using a domain that can't be provided with a SSL cert as of now.

We've configured Weblate according to the documentation and provided the required credentials in the `settings.py` so that we can see the GitLab integration within the VCS dropdown.

After changing some translations and trying to push them we can see that Weblate tries to access the GitLab API but fails due to it not supporting HTTPS.

We've tried the component settings using SSH url as well as the HTTP clone URL provided from GitLab including the credentials as stated [here](https://docs.weblate.org/en/latest/vcs.html#https-repositories) but without https:

```
http://<gitlab-user>:<gitlab-token>@<gitlab-http-clone-url>
```

We've also tried to configure the credentials used by Weblate to include `http://` which unfortunately results in an `hostname` not found error as this code part (obviously) fails then
https://github.com/WeblateOrg/weblate/blob/ae32bbcff2855e76189ad96b743532c8080b9eec/weblate/vcs/git.py#L781

As far as I can understand from the code this seems to be because of the hardcoded `https://` at https://github.com/WeblateOrg/weblate/blob/ae32bbcff2855e76189ad96b743532c8080b9eec/weblate/vcs/git.py#L1275

Is this assumption correct that there is no way to configure Weblate to use the API via HTTP instead? Would this be a viable setting?

Thanks a lot,
Patrick

**Comment by nijel** [comment, marked as answer]
Only HTTPS is supported, and I don't think it's a good idea to expose API keys over insecure HTTP.

**Reply by pschneider** [reply]
@nijel Thank you for confirming this. Appreciate your quick help 🙏  

Of course, I totally agree with you.
As this whole setup is only available within the company network we don't think that would be a huge problem. I was asking if this might be a viable option due to other companies maybe having similar setups (still).

We would postpone this until we can make the whole infrastructure available on https.

**Reply by nijel** [reply]
I think adding `protocol` (defaulting to `https`) to the [GITLAB_CREDENTIALS](https://docs.weblate.org/en/latest/admin/config.html#gitlab-credentials) would be an acceptable approach without complicating the code base too much.

**Reply by pschneider** [reply]
I think that would solve our use case. If that is indeed not a heavy change on the codebase of Weblate it would be great to have as an option. We could then use this integration already while we're planning the migration to the full https infrastructure.

**Reply by nijel** [reply]
https://github.com/WeblateOrg/weblate/issues/9179 was created to track this feature request.
